packages/labs/virtualizer/src/test/scenarios/element-and-directive-parity.test.ts

```diff
@@ -140,70 +140,56 @@ describe('lit-virtualizer and virtualize directive', () => {
     uvd.selected = selected;
 
     // Wait for events from initial render to fire.
-    await until(() => ulv.rangeChangedEvents.length > 0);
-    await until(() => uvd.rangeChangedEvents.length > 0);
-    await until(() => ulv.visibilityChangedEvents.length > 0);
-    await until(() => uvd.visibilityChangedEvents.length > 0);
+    await until(() => expect(ulv.rangeChangedEvents.length).to.equal(3));
```

Comment:
When a condition function with an `expect` is passed in, we're expecting it to actually throw rather than return a falsy value so we're not going to have the benefit of the stack trace of `until()`. Would it make sense to just use `eventually()` directly instead for all of these then?

If we otherwise anticipate/encourage usage of `expect` with `until`, I think there should be mention of that for in `until()`'s jsdoc.

Reply:
So the `until` stack trace code was actually a hack to get something useful to display, namely the source code line that expressed the condition.  It's less useful than what the expectation error emits.  But you're right that the usage of this here is kind of misleading in the sense that the expectation object itself is a truthy thing, being just object or whatever, but its existence doesn't "mean" anything truthy in the semantic sense of until.

The subtle thing about `eventually` is that you still want to return something truthy so you don't have a timeout condition while it's waiting for something truthy.  That means that for concision you're still likely to wind up using the expectation object directly as the return value like:

```ts
await eventually(() => expect(something).to.be.whatever);
```

For example, I beat myself up trying to figure out why I couldn't bundle expectations like this until I realized I needed to return something true at the end or it would timeout at 1000ms and then if i did it more than 2 times per test the default 2000ms mocha timeout would fail my test:

```ts
await eventually(() => {
   expect(x).to.equal(1);
   expect(y).to.equal(2);
});

In reality what I was trying to do was have a block of code that would run until it didn't cause an exception.  But I didn't make a semantic for that yet.  Could create a new function for that though, perhaps `pass`?   Like:

```ts
await pass(() => expect(x).to.equal(1));

await pass(() => {
  expect(x).to.equal(1)
  expect(y).to.equal(2)
});
```

This semantic basically says it will continue as long as no error is thrown and it will keep retrying until it passes or the timeout is reached.

Either that or I could redefine `eventually` to perform that function, I just don't know if this meaning is implied in that word?

Reply:
Aha, right. I do like `pass` or `passing` for the semantic that only checks for exceptions thrown and not a boolean condition.

This would leave `until`/`eventually` generally to be used for asynchronously grabbing truthy values.

Reply:
On quite-a-bit-more further thought, as written, `eventually` doesn't really support any direct use cases that are different than `until` since it still will timeout until it gets something truthy.  Its name is also vague, since eventually and until kind of mean the same thing.  I think I'll just have `pass` and `until` which are more clearly semantically different, even if not immediately obvious what they are as functions unless one sees them in action or reads their docs.

Reply:
Okay code pushed and all tests changed over to reflect the new `until` and `pass` reality.